amino.go

```diff
@@ -197,7 +197,7 @@ func (cdc *Codec) MarshalBinaryBare(o interface{}) ([]byte, error) {
 
 	// Encode Amino:binary bytes.
 	var bz []byte
-	buf := new(bytes.Buffer)
+	buf := bytes.NewBuffer([]byte{})
```

Comment:
I don't think this is necessary, fyi.  From the docs,
"The zero value for Buffer is an empty buffer ready to use."

Reply:
True, it is ready to use but the returned bytes will differ from what go-protobuf returns (`[]byte(nil)` vs `[]byte{}`). Alternatively, we can make the tests treat these cases as equal instead of modifying it here.